source/containers.tex

```diff
@@ -14292,6 +14384,1710 @@
 As if by \tcode{x.swap(y)}.
 \end{itemdescr}
 
+\rSec2[flatmap]{Class template \tcode{flat_map}}
+
+\rSec3[flatmap.overview]{Overview}
+
+\pnum
+\indexlibrary{\idxcode{flatmap}}%
+A \tcode{flat_map} is a container adaptor
+that provides an associative container interface
+that supports unique keys
+(i.e., contains at most one of each key value) and
+provides for fast retrieval of values of another type \tcode{T}
+based on the keys.
+\tcode{flat_map} supports iterators
+that meet the \oldconcept{InputIterator} requirements and
+model the
+\libconcept{random_access_iterator} concept\iref{iterator.concept.random.access}.
+
+\pnum
+A \tcode{flat_map} meets all of the requirements
+for a container\iref{container.reqmts} and
+for a reversible container\iref{container.rev.reqmts},
+plus the optional container requirements\iref{container.opt.reqmts}.
+\tcode{flat_map} meets the requirements of
+an associative container\iref{associative.reqmts},
+except that:
+\begin{itemize}
+\item
+it does not meet the requirements related to node handles\iref{container.node},
+\item
+it does not meet the requirements related to iterator invalidation, and
+\item
+the time complexity of the operations that insert or erase a single
+element from the map is linear, including the ones that take an insertion
+position iterator.
+\end{itemize}
+\begin{note}
+A \tcode{flat_map} does not meet the additional requirements of
+an allocator-aware container\iref{container.alloc.reqmts}.
+\end{note}
+
+\pnum
+A \tcode{flat_map} also provides most operations
+described in \ref{associative.reqmts} for unique keys.
+This means that a \tcode{flat_map} supports
+the \tcode{a_uniq} operations in \ref{associative.reqmts},
+but not the \tcode{a_eq} operations.
+For a \tcode{flat_map<Key,T>}
+the \tcode{key_type} is \tcode{Key} and
+the \tcode{value_type} is \tcode{pair<Key, T>}.
+
+\pnum
+Descriptions are provided here only for operations on \tcode{flat_map} that
+are not described in one of those sets of requirements or for operations where
+there is additional semantic information.
+
+\pnum
+A \tcode{flat_map} maintains the following invariants:
+\begin{itemize}
+\item
+it contains the same number of keys and values;
+\item
+the keys are sorted with respect to the comparison object; and
+\item
+the value at offset \tcode{off} within the value container is
+the value associated with the key at offset \tcode{off}
+within the key container.
+\end{itemize}
+
+\pnum
+If any member function in \ref{flatmap.defn} exits via an exception
+the invariants are restored.
+\begin{note}
+This can result in the \tcode{flat_map}'s being emptied.
+\end{note}
+
+\pnum
+Any sequence container\iref{sequence.reqmts} \tcode{C}
+supporting \oldconcept{RandomAccessIterator} can be used
```

Comment:
This must be referring to `C`'s associated iterator type(s). How can this be improved? For reference:

> Any sequence container with random access iterator and supporting operations front(), push_­back() and pop_­back() can be used to instantiate priority_­queue[.](https://eel.is/c++draft/priority.queue#priqueue.overview-1.sentence-1)
> -- https://eel.is/c++draft/priority.queue#priqueue.overview-1.sentence-1

Reply:
@jwakely, @CaseyCarter: Any thoughts? I don't think this is broken enough to warrant an immediate fix here, but I'm happy to entertain editorial improvement requests.